### PR TITLE
Bug 1177798 - Only cache a dictionary of user data instead of full ORM objects.

### PR DESCRIPTION
**Pull request by jezdez**
This should reduce the amount spent in Python a bunch.

### CODE REVIEW

File: kuma/wiki/jobs.py [hunk 1]
```diff
@@ -61,6 +61,7 @@ class DocumentContributorsJob(KumaJob):
     """
     lifetime = 60 * 60 * 6
     refresh_timeout = 30
+    version = 2
```
Comment (jezdez) [inline kuma/wiki/jobs.py]: Increasing the job version here to make sure we invalidate things on deploy.

File: kuma/wiki/templates/wiki/document.html [hunk 2]
```diff
@@ -180,8 +180,9 @@
         {% endif %}
 
         {% if waffle.flag('top_contributors') %}
+            {% set contributors = document.get_contributors() %}
```
Comment (groovecoder) [inline kuma/wiki/templates/wiki/document.html]: [The call to `user_list` below here](https://github.com/mozilla/kuma/blob/bug1177798-2/kuma/wiki/templates/wiki/document.html#L359) expects the `contributors` object to be a QuerySet of users [to use the `get_absolute_url()` method](https://github.com/mozilla/kuma/blob/bug1177798-2/kuma/users/helpers.py#L69-L70). Causes this traceback on viewing a document:
```
Traceback (most recent call last):
  File "/usr/lib/python2.7/wsgiref/handlers.py", line 85, in run
    self.result = application(self.environ, self.start_response)
  File "/home/vagrant/src/vendor/src/django/django/contrib/staticfiles/handlers.py", lin
e 64, in __call__
    return self.application(environ, start_response)
  File "/home/vagrant/src/vendor/src/django/django/core/handlers/wsgi.py", line 187, in
__call__
    response = self.get_response(request)
  File "/home/vagrant/src/vendor/src/django/django/core/handlers/base.py", line 199, in
get_response
    response = self.handle_uncaught_exception(request, resolver, sys.exc_info())
  File "/home/vagrant/src/vendor/src/django/django/core/handlers/base.py", line 111, in
get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/vagrant/src/vendor/src/django/django/db/transaction.py", line 394, in inne
r
    return func(*args, **kwargs)
  File "/home/vagrant/src/vendor/src/django/django/views/decorators/csrf.py", line 57, i
n wrapped_view
    return view_func(*args, **kwargs)
  File "/home/vagrant/src/vendor/src/django/django/views/decorators/http.py", line 41, i
n inner
    return func(request, *args, **kwargs)
  File "/home/vagrant/src/kuma/wiki/decorators.py", line 32, in _added_header
    response = func(request, *args, **kwargs)
  File "/home/vagrant/src/kuma/authkeys/decorators.py", line 33, in process
    return func(request, *args, **kwargs)
  File "/home/vagrant/src/kuma/wiki/decorators.py", line 107, in process
    return func(request, *args, **kwargs)
  File "/home/vagrant/src/vendor/src/django/django/views/decorators/http.py", line 148,
in inner
    response = func(request, *args, **kwargs)
  File "/home/vagrant/env/local/lib/python2.7/site-packages/newrelic-2.50.0.39/newrelic/
api/function_trace.py", line 104, in literal_wrapper
    return wrapped(*args, **kwargs)
  File "/home/vagrant/src/kuma/wiki/views.py", line 573, in document
    response = render(request, 'wiki/document.html', context)
  File "/home/vagrant/src/vendor/src/django/django/shortcuts.py", line 50, in render
    return HttpResponse(loader.render_to_string(*args, **kwargs),
  File "/home/vagrant/src/vendor/src/django/django/template/loader.py", line 178, in ren
der_to_string
    return t.render(context_instance)
  File "/home/vagrant/src/vendor/src/jingo/jingo/__init__.py", line 195, in render
    return super(Template, self).render(context_dict)
  File "/home/vagrant/env/local/lib/python2.7/site-packages/jinja2/environment.py", line
 969, in render
    return self.environment.handle_exception(exc_info, True)
  File "/home/vagrant/env/local/lib/python2.7/site-packages/jinja2/environment.py", line 742, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/vagrant/src/kuma/wiki/templates/wiki/document.html", line 41, in top-level
 template code
    {% set content_class = 'column-main' %}
  File "/home/vagrant/src/kuma/wiki/templates/wiki/base.html", line 14, in top-level template code
    {% set scripts = ('wiki',) %}
  File "/home/vagrant/src/templates/base.html", line 149, in top-level template code
    {% block content %}{% endblock %}
  File "/home/vagrant/src/kuma/wiki/templates/wiki/document.html", line 358, in block "content"
    {% include "wiki/includes/document_tag.html" %}
  File "/home/vagrant/src/kuma/users/helpers.py", line 70, in user_list
    u in users])
AttributeError: 'dict' object has no attribute 'get_absolute_url'
```

Comment (jezdez) [inline kuma/wiki/templates/wiki/document.html]: Fixed by refactoring the `user_list` helper into a Jinja2 macro

Comment (groovecoder) [inline kuma/wiki/templates/wiki/document.html]: Looks good now.

**Comment by groovecoder** [comment]
r+ when Travis approves

**Comment by jezdez** [comment]
sigh, not sure what's wrong with travis

**Comment by jezdez** [comment]
Travis is fixed again, seems to have been a snafu

**Pull request merged**